rust/noosphere/src/ffi/authority.rs

```diff
@@ -0,0 +1,335 @@
+            Ok(name) => async_runtime.spawn_blocking(move || callback(context, None, Some(name))),
+            Err(error) => async_runtime.spawn_blocking(move || {
+                callback(context, Some(NoosphereError::from(error).into()), None)
+            }),
+        };
+    });
+}
+
+/// @memberof ns_sphere_t
+///
+/// Get the DID associated with the given authorization. The authorization should be
+/// provided as a base64-encoded CIDv1 string.
+///
+/// The callback arguments are (in order):
+///
+///  1. The context argument provided in the original call to
+///     ns_sphere_authority_authorize
+///  2. An owned pointer to an ns_error_t if there was an error, otherwise NULL
+///  3. An owned pointer to constant string (the authorized DID) if the call
+///     was successful, otherwise NULL
+pub fn ns_sphere_authority_authorization_identity(
+    noosphere: &NsNoosphere,
+    sphere: &NsSphere,
+    authorization: char_p::Ref<'_>,
+    context: Option<repr_c::Box<c_void>>,
+    callback: extern "C" fn(
+        Option<repr_c::Box<c_void>>,
+        Option<repr_c::Box<NsError>>,
+        Option<char_p::Box>,
+    ),
+) {
+    let sphere = sphere.inner().clone();
+    let authorization = authorization.to_string();
+    let async_runtime = noosphere.async_runtime();
+
+    noosphere.async_runtime().spawn(async move {
+        let result = async {
+            let link = Link::<Jwt>::from(Cid::try_from(authorization)?);
+
+            let ucan = Authorization::Cid(link.into())
+                .as_ucan(sphere.sphere_context().await?.db())
+                .await?;
+
+            let identity: char_p::Box = ucan.audience().to_string().try_into()?;
+
+            Ok(identity) as Result<_, anyhow::Error>
+        }
+        .await;
+
+        match result {
+            Ok(identity) => {
+                async_runtime.spawn_blocking(move || callback(context, None, Some(identity)))
+            }
+            Err(error) => async_runtime.spawn_blocking(move || {
+                callback(context, Some(NoosphereError::from(error).into()), None)
+            }),
+        };
+    });
+}
+
+#[ffi_export]
+/// @memberof ns_sphere_t
+///
+/// Given a sphere root credential recovery mnemonic, get back a ns_sphere_t
+/// with an escalated credential that allows signing changes as the sphere
+/// itself.
+///
+/// This function call will return an error if the mnemonic is invalid or if the
+/// credential it represents is not the sphere's key.
+///
+/// The callback arguments are (in order):
+///
+///  1. The context argument provided in the original call to
+///     ns_sphere_authority_authorize
+///  2. An owned pointer to an ns_error_t if there was an error, otherwise NULL
+///  3. An owned pointer to an ns_sphere_t with the root sphere credential,
+///     otherwise NULL
+#[allow(clippy::type_complexity)]
+pub fn ns_sphere_authority_escalate(
+    noosphere: &NsNoosphere,
+    sphere: &mut NsSphere,
+    mnemonic: char_p::Ref<'_>,
```

Comment:
Is mnemonicification of private keys generally applicable across key types? If not, perhaps the public interface for this should simply be "private key" (string), with some mnemonic to/from helpers. Something to consider in the future